lib/msf/core/handler.rb

```diff
@@ -245,11 +245,7 @@ def register_session(session)
 
     # Call the handler's on_session() method
     on_session(session)
-
-    # Process the auto-run scripts for this session
-    if session.respond_to?('process_autoruns')
-      session.process_autoruns(datastore)
-    end
+    session.bootstrap(datastore)
```

Comment:
This is a great example of synchrony problems in distributed systems - my sense is that a similar condition occurs at the channel/socket tier when running portscans and the like via meterp pivots and something that isn't expected gets brought into the pipeline, and then something that is does not (resulting in that awesome nil call on the channel, which seems to be gone with the timeouts restored, indicating we've hidden the problem again).
With a large number of targets in SDNs these days, the value of session pivots is growing, so we probably want to find some way to check client/server behaviors in rspec or another test framework to ensure that both sides are playing the same tune.